File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const partnerIndexes = {};
  
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    
    if (partnerIndexes.hasOwnProperty(num)) {
      return [partnerIndexes[num], i];
    }
    
    partnerIndexes[target - num] = i;
  }
  
  return null;
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        partner_indexes = {}
        
        for i, num in enumerate(nums):
            if num in partner_indexes:
                return [partner_indexes[num], i]
            
            partner_indexes[target - num] = i
        
        return None